### ISSUE TITLE
Styles enhancements poll

### ISSUE THREAD

**Issue by delister** [Nheko-Reborn/nheko#96]
I think you need to increase the vertical indent between posts. And, as a result, also increase the indent between other people's posts.

If the message is short, then the distance between the messages of two people is enough and looks harmonious. But if the message is long, then the indentation is not enough, and the message of the next person begins to be perceived as a continuation of the current one. In addition, when the message is long and another message from the same person immediately falls under it, such a text is difficult to read. The upper line “squeezes”, and the gaze periodically rises.

In addition, I am convinced that it will look better.

This is a subjective opinion, it is necessary that others also speak out about this. Maybe it's possible to put it in a configuration file? So that everyone can adjust the indentation for themselves.

## Name colors
Another improvement that I see is a change in the color of the names of the interlocutors. Now a case is possible when the name is difficult to read, due to the low contrast with the background. Maybe it's possible to change the color matching algorithm? or give the opportunity to specify a range of colors. Ideally, I would like to be able to include only 2 colors: one for my name, the other for all other names.

**Comment by deepbluev7** [comment]
## Message spacing

I think this one is very subjective. In my opinion, if you have a problem staying in one line when reading, this will still be an issue with longer messages that take multiple lines or maybe entire paragraphs. Increasing the spacing between messages wouldn't really solve that. I think limiting the line length or increasing line spacing would help that more. In my opinion there isn't really a difference of sending one message with 2 lines or two messages with one line, especially when it is pretty hard at the moment to send an image inside a normal text message.

A different approach to go at this, would be to add a background to each message, like other messengers do. This helps a lot more to distinct different messages and would also tie in with the second one of your points, as you could have two background colors. One for your messages, one for other messages.

## Name colors

If the name colors don't have high enough contrast, we may need to look at the color algorithm again. It uses a contrast threshold, when generating the color, but it could be, that we are passing the wrong background color in or that the threshold is too low/buggy. Do you have an example, of where that is the case?

Also I'm currently experimenting with a timeline view written in qml in !91. That may make it easier to adapt any of your suggestions and make them configurable, but it also looks a bit different already. Also could you provide the platform your on and what theme you are using? It could also be, that the contrast is system/theme specific.

**Comment by delister** [comment]
## Message spacing
Yes, you are right, this is subjective, that's why I wanted to discuss and define it together. I also agree that line spacing is a more correct solution than indentation. But if you increase the line spacing, you will also have to increase the indentation between messages from different people, so that they visually separate. I mean, it will not be enough to change just the line spacing.

I switched from Riot and this is the first thing that caught my eye, maybe it's a matter of habit.

Regarding the background, if you mean something like a telegram, then I personally do not really like it, maybe most will like it more on the contrary.

## Name colors

I use gentoo. Theme GTK2 ARC. Nheko Light theme.

For example, here it’s not convenient for me to read the name: xshram_original
![Image of Yaktocat](https://cloud.skoryh.com/s/oTLQdF48bPABA8p/preview)

and here: swedneck:permaweb.io
![Image of Yaktocat](https://cloud.skoryh.com/s/ABMbAXJGW4H5QsF/preview)

**Comment by deepbluev7** [comment]
## Message spacing

Of course an increased line spacing would also need an increased message spacing. But I wonder, do you think it is important to actually distinguish different messages? Usually I only need that, if the messages are sent on different days or id they are sent by different people, which is already taken care of by the date separators and user names before batches of messages. Currently they are inserted only every day or if the author changes, but that is easy to tweak.

I've always used the compact layout in Riot, which is a lot more similar to how nheko does it currently. We could maybe have a toggle like that, which changes avatar size, message spacing and maybe line spacing. We could also add visual separators in the non compact version between messages, i.e. a bit lighter/darker backround for each message and leaving the space between each message in the window colour. I.e.:

|property|compact layout|non-compact layout|
|-----|----|-----|
|line spacing|default|1.2x default|
|avatar size|32|48|
|message spacing|normal line spacing/none|height of one text line|
|color of space between messages|no space -> none|a bit lighter/darker than current background color? Username color? Something that looks good.|

## Name colors

Wow, those look amazingly hard to read! I haven't had that issue yet, but I'm mostly using the dark theme. Maybe our contrast code is broken for light backgrounds or maybe you can't make colours like yellow readable on light backgrounds. I'll try to verify that.

**Comment by delister** [comment]
## Name colors
Yes, the dark theme does not have such problems. While I will use it, thanks.

## Message spacing
I don’t think it’s very important (to distinguish between messages), but I think that it will increase comfort, and to a greater extent it has a purely cosmetic function (again in my opinion).

And this idea with switching display modes, I really like. The only thing, I think that you should not increase the avatar, what do you think? Now he has opened a riot and watched how their mode switching works (compact and ordinary). In compact mode, only the indent between messages is reduced. Line height, avatar and everything else remains the same.

I also want to give an example of large messages from riot and nheko
Nheko:
![Nheko](https://cloud.skoryh.com/s/gK2j4Arm2YAnrSR/preview)

Riot:
![Nheko](https://cloud.skoryh.com/s/zkznLcdq5R73BfC/preview)

This is what I said at the beginning that with a small line height, the text merges and is not very convenient to read. (The example is not entirely clean, because in riot indents between paragraphs are still implemented, but I think you understand what I mean)

I also think that in non-compact mode you should not select messages using the background, only indentation will suffice. In my opinion this will be redundant.

**Comment by rnhmjoj** [comment]
Something that I don't like particularly are the new "options" and "reply" buttons that are always visible on the right of any message. IMHO they clutter interface and should only appear when then message is hovered/selected.

**Comment by deepbluev7** [comment]
Yeah, that may be reasonable. It could also give us more space for the messages, although then it could also overlap with the message... Hm, I'll need to experiment with that.

**Comment by InnayTool** [comment]
I think the spacing problem is not only in between the messages. The whole application could look and feel better if there would be added more space without "content", called [negative space](https://en.wikipedia.org/wiki/Negative_space): 
> Considering and improving the balance between negative space and positive space in a composition is considered by many to enhance the design. This basic, but often overlooked, principle of design gives the eye a "place to rest," increasing the appeal of a composition through subtle means.

**Comment by vachan-maker** [comment]
The text in the white boxes in the settings menu when dark mode is barely visible. Use of a different colour will probably solve the issue.

**Comment by duarm** [comment]
too many badges (edited, sent, trusted)
![image](https://user-images.githubusercontent.com/26150581/160416974-e11bc53e-3713-4d7f-8b0f-d4086a9d37bc.png)
It would be nice to move all of those, including the date, to the name row perhaps, and make the text row extend all the way to the window bounds. The per text info could be displayed when hovering and specific message, or when right clicking it.

**Comment by deepbluev7** [comment]
If you use the bubbles layout, they are all below the message. I'm not that enthusiastic about hiding them, since I like seeing that info at a glance.

**Comment by duarm** [comment]
> If you use the bubbles layout, they are all below the message

I don't like the bubble layout, but even the bubble layout does not extend to the edge of the window

>  I'm not that enthusiastic about hiding them, since I like seeing that info at a glance.

it could still be visible, but it could be displayed on that unused space instead of shrinking the message.

**Comment by deepbluev7** [comment]
There is no unused space with my font size though:

![grafik](https://user-images.githubusercontent.com/16312324/160507175-dd552edf-877f-4113-b99d-e52e45f020ac.png)

**Comment by duarm** [comment]
yeah, this change would require changes to the message layout, but the goal is to extend the message as far as possible to the edge of the screen, the badges just takes too much space for information I don't need most of the time, or I never need (sent/read badge). 

The problem gets worse when on small monitors or on narrow windows (mobile, tilling window managers)

**Comment by chaserene** [comment]
the line spacing is just too small. it makes the text quite hard to read, to the point I have an issue using the application. could you please make this a parameter that can be calibrated in Settings?

a bit larger space to separate messages would also do a lot for readability.
